proxy/frontend/type/postgresql/src/main/java/org/apache/shardingsphere/proxy/frontend/postgresql/command/query/extended/parse/PostgreSQLComParseExecutor.java

```diff
@@ -87,6 +93,13 @@ private SQLParserEngine createShardingSphereSQLParserEngine(final String databas
         return sqlParserRule.getSQLParserEngine(protocolType.getTrunkDatabaseType().orElse(protocolType));
     }
     
+    private String escape(final SQLStatement sqlStatement, final String sql) {
+        if (sqlStatement instanceof DMLStatement) {
+            return sql.replace("?", "??");
+        }
+        return sql;
+    }
```

Comment:
I found the doc about escape `?` in the following link. I think the escape logic could be merged into `convertSQLToJDBCStyle` and invoke the method with `convertSQLToJDBCStyle(Collections.emptyList(), sql)` if parameter placeholder is absent.
https://jdbc.postgresql.org/documentation/query/#using-the-statement-or-preparedstatement-interface

Reply:
i found, if  this escape  moved to    convertSQLToJDBCStyle,   the sqlStatement's parameter mark indexes will be missed up